File: net/Azure.Storage.Blobs.PerfStress/UploadTest.cs
using Azure.Storage.Blobs.PerfStress.Core;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Storage.Blobs.PerfStress
{
    public class UploadTest : RandomBlobTest<StorageTransferOptionsOptions>
    {
        public UploadTest(StorageTransferOptionsOptions options) : base(options)
        {
        }

        public override void Run(CancellationToken cancellationToken)
        {
            using var stream = RandomStream.Create(Options.Size);
            BlobClient.Upload(stream, transferOptions: Options.StorageTransferOptions, cancellationToken: cancellationToken);
        }

        public override async Task RunAsync(CancellationToken cancellationToken)
        {
            using var stream = RandomStream.Create(Options.Size);
            await BlobClient.UploadAsync(stream, transferOptions: Options.StorageTransferOptions,  cancellationToken: cancellationToken);
        }
    }
}

File: java/azure-storage-blob-perfstress/src/main/java/com/azure/storage/blob/perfstress/UploadTest.java
package com.azure.storage.blob.perfstress;

import java.io.IOException;

import com.azure.perfstress.RandomFlux;
import com.azure.perfstress.RandomStream;
import com.azure.perfstress.SizeOptions;
import com.azure.storage.blob.models.ParallelTransferOptions;
import com.azure.storage.blob.perfstress.core.RandomBlobTest;

import reactor.core.publisher.Mono;

public class UploadTest extends RandomBlobTest<SizeOptions> {

    public UploadTest(SizeOptions options) {
        super(options);
    }

    @Override
    public void Run() {
        try {
            _blobClient.getBlockBlobClient().upload(RandomStream.create(Options.Size), Options.Size);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public Mono<Void> RunAsync() {
        return _blobAsyncClient.upload(RandomFlux.create(Options.Size), new ParallelTransferOptions()).then();
    }
}